### ISSUE TITLE
`JsonType.As.EXISTING_PROPERTY` cannot handle multiple types with same class

### ISSUE THREAD

**Issue by natnan** [FasterXML/jackson-databind#716]
This doesn't seem to be a problem when using `JsonType.As.PROPERTY`, so I didn't consider it as an illegal thing to do.

```Java
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", include = JsonTypeInfo.As.EXISTING_PROPERTY)
@JsonSubTypes({
    @JsonSubTypes.Type(value = SubClass1.class, name = "1"),
    @JsonSubTypes.Type(value = SubClass1.class, name = "1,5"),
    @JsonSubTypes.Type(value = SubClass2.class, name = "2"),
    @JsonSubTypes.Type(value = SubClass2.class, name = "2.25"),
    @JsonSubTypes.Type(value = SubClass2.class, name = "2.5"),
    @JsonSubTypes.Type(value = SubClass2.class, name = "2.75")
})
```
And this is the exception message I get:
```
Could not resolve type id 'Create' into a subtype of [simple type, class com.test.MySuperTestPojo]: known type ids = [MySuperTestPojo, 1, 2]
```

I'll try to write a unit test when I have time later

P.S.: (If I use `As.Property`, it ignores `JsonPropertyOrder`, and I think it serializes the properties in a somewhat random order. Another issue?)

**Comment by natnan** [comment]
Actually, I'm having the same issue with `As.PROPERTY` now as well. Maybe it's something else.. I'll update this when i know more.

**Comment by cowtowncoder** [comment]
I think this is #312, which will be fixed in 2.6.0. So you may want to try 2.6.0-SNAPSHOT to see if that might help.

**Comment by cowtowncoder** [comment]
As to ordering: it should retain ordering (minus actual property used which will be serialized as the first thing). Although if you have existing property you want to bind to, and you want to specify its ordering, `As.EXISTING_PROPERTY` should retain ordering of even that one property.

If ordering is otherwise messed up, that would be a bug (type id and object id are exceptions to the rule).

**Comment by natnan** [comment]
Yes, looks like #312 is the issue which I didn't come across because I thought it was a problem with `As.EXISTING_PROPERTY` and that's what I've searched before posting. I've tried with 2.6.0-SNAPSHOT and it seems to be working fine.

As for the ordering, yes that was my expectation which wasn't met. I've tried to reproduce it with unit tests on 2.6.0 but didn't manage to. Maybe I was doing something wrong and misread things; or some chained actions were causing it as lots of object mapping occurs before it happens in the application tests.
